lib/box-node-sdk.js

```diff
@@ -118,6 +118,67 @@ BoxSDKNode.collaborationRoles = BoxSDKNode.prototype.collaborationRoles = BoxCli
 BoxSDKNode.CURRENT_USER_ID = BoxSDKNode.prototype.CURRENT_USER_ID = BoxClient.prototype.CURRENT_USER_ID;
 
 /**
+ * Gets the BoxSDKNode instance by passing boxAppSettings json downloaded from the developer console.
+ *
+ * @param {object} appConfig boxAppSettings json is passed as parameter.
+ * @returns {BoxSDKNode} the BoxSDKNode instance
+ */
+BoxSDKNode.getPreconfiguredInstance = function(appConfig) {
+	if (typeof appConfig.boxAppSettings !== 'object') {
+		throw new Error('Configuration does not include boxAppSettings object.');
+	}
+
+	var boxAppSettings = appConfig.boxAppSettings;
+	var webhooks = boxAppSettings.webhooks;
+	if (typeof webhooks === 'object') {
+		Webhooks.setSignatureKeys(webhooks.primaryKey, webhooks.secondaryKey);
+	}
+
+	var params = {};
+	if (typeof boxAppSettings.clientID === 'string') {
+		params.clientID = boxAppSettings.clientID;
+	}
+
+	if (typeof boxAppSettings.clientSecret === 'string') {
+		params.clientSecret = boxAppSettings.clientSecret;
+	}
+
+	// Assign publicKeyID to keyID
+	if (typeof boxAppSettings.appAuth === 'object') {
+
+		var appAuth = {};
+		var keyID = boxAppSettings.appAuth.publicKeyID;
+		if (typeof keyID === 'string') {
+			appAuth.keyID = keyID;
+		}
+
+		appAuth.privateKey = boxAppSettings.appAuth.privateKey;
+
+		var passphrase = boxAppSettings.appAuth.passphrase;
+		if (typeof passphrase === 'string') {
+			appAuth.passphrase = passphrase;
+		}
+	}
+	params.appAuth = appAuth;
+
+	if (typeof boxAppSettings.enterpriseID === 'string') {
+		params.enterpriseID = boxAppSettings.enterpriseID;
+	}
+
+	return new BoxSDKNode(params);
+};
+
+/**
+ * Configures additional properties passed as parameter.
```

Comment:
This description is a bit hard to understand.  Perhaps something like "Updates the SDK configuration with new parameters" would be clearer